ts/WoltLabSuite/WebComponent/woltlab-core-pagination.ts

```diff
@@ -0,0 +1,318 @@
+      if (this.page === this.count) {
+        return;
+      }
+
+      const div = document.createElement("div");
+      div.classList.add(`${this.#className}__next`);
+
+      const button = this.#getButtonElement(this.page + 1);
+      if (button instanceof HTMLAnchorElement) {
+        button.rel = "next";
+      }
+      button.title = window.WoltLabLanguage.getPhrase("wcf.global.page.next");
+      button.classList.add("jsTooltip");
+      div.append(button);
+
+      const icon = document.createElement("fa-icon");
+      icon.setIcon("arrow-right");
+      icon.ariaHidden = "true";
+      button.append(icon);
+
+      return div;
+    }
+
+    #getButtonElement(page: number): HTMLAnchorElement | HTMLButtonElement {
+      let button: HTMLAnchorElement | HTMLButtonElement;
+      const url = this.getLinkUrl(page);
+
+      if (url) {
+        button = document.createElement("a");
+        button.href = url;
+      } else {
+        button = document.createElement("button");
+        button.type = "button";
+
+        if (this.page === page) {
+          button.disabled = true;
+        } else {
+          button.addEventListener("click", () => {
+            this.#switchPage(page);
+          });
+        }
+      }
+
+      button.classList.add(`${this.#className}__link`);
+
+      return button;
+    }
+
+    #getLinkItem(page: number): HTMLLIElement {
+      const li = document.createElement("li");
+      li.classList.add(`${this.#className}__item`);
+
+      const button = this.#getButtonElement(page);
+      button.ariaLabel = window.WoltLabLanguage.getPhrase("wcf.page.pageNo", { pageNo: page });
+      if (page === this.page) {
+        button.setAttribute("aria-current", "page");
+        button.classList.add(`${this.#className}__link--current`);
+      }
+      button.textContent = page.toString();
+      li.append(button);
+
+      return li;
+    }
+
+    /**
+     * Generates page numbers that are adjacent to the current page.
+     */
+    #getLinkItems(): HTMLLIElement[] {
+      const items: HTMLLIElement[] = [];
+
+      let start: number;
+      let end: number;
+
+      if (getMediaQueryScreenXs().matches) {
+        // On small devices show only the first, last and current
+        // page number.
+        start = this.page;
+        end = this.page;
+      } else {
+        // Avoid generating an ellipsis which only skips a single
+        // page number. Instead of `1 ⋯ 3` this code will always
+        // generate `1 2 3`, but on page 5 it will be `1 ⋯ 4 5`.
+        start = this.page - 1;
+        if (start === 3) {
+          start--;
+        }
+        end = this.page + 1;
+        if (end === this.count - 2) {
+          end++;
+        }
+      }
+
+      for (let i = start; i <= end; i++) {
+        if (i <= 1 || i >= this.count) {
+          continue;
+        }
+
+        items.push(this.#getLinkItem(i));
+      }
+
+      return items;
+    }
+
+    #getEllipsesItem(): HTMLLIElement {
```

Comment:
```suggestion
    #getEllipsisItem(): HTMLLIElement {
```
?

Also applies to the `className`.